### PR TITLE
[NR-102311] Memory issue fix

### PR DESCRIPTION
**Pull request by ywang-nr**
1. Move waitablerunner outside the loop and reset signal flag
2. Turn on ANRMonitor
3. Added logs for future debugging reference

### CODE REVIEW

File: agent-ndk/src/main/java/com/newrelic/agent/android/ndk/ANRMonitor.kt [hunk 1]
```diff
@@ -45,7 +45,11 @@ open class ANRMonitor {
                         return@Runnable
                     }
 
-                    runner.wait(ANR_TIMEOUT)
+                    try {
+                        Thread.sleep(ANR_TIMEOUT)
+                    } catch (e: Exception) {
+                        e.printStackTrace()
```
Comment (cthomas-newrelic) [inline agent-ndk/src/main/java/com/newrelic/agent/android/ndk/ANRMonitor.kt:51]: This is good to know, but shoudl anything else be done in the monitor is a sleep interruption occurs?

**Review by cthomas-newrelic — CHANGES_REQUESTED** [review]
Other than the try/cathc I don't see anything that would address a memory leak.

**Comment by ywang-nr** [comment]
> Other than the try/catdh I don't see anything that would address a memory leak.

So first I have discovered that `runner.wait(TIMEOUT)` doesn't take affects, while the loop itself keeps spinning, supposedly gc should take affect to clean up `WaitableRunner` but it didn't.

After I change it into `thread.sleep`, the memory leak is no longer happening from the profiler view. The theory I came up with is by put the loop thread into sleep allows other process takes place, aka gc is kicking in to clean up `WaitableRunner`. 

Profile compare as below:
Screenshot 2023-08-24 at 5 13 47 PM

**Comment by cthomas-newrelic** [comment]
1. However, the `wait(TIMEOUT)` waits on notification from the runner which may be up to the TIMEOUT duration, or more usually right away. `sleep()` will cause the thread to block every loop, which is not optimal. I'd like to know why, as you say,  runner.wait(TIMEOUT) doesn't take affect. Under normal conditions, the runner should be signaled immediately, and not wait.

File: agent-ndk/src/main/java/com/newrelic/agent/android/ndk/ANRMonitor.kt [hunk 2]
```diff
@@ -48,8 +48,12 @@ open class ANRMonitor {
                     runner.wait(ANR_TIMEOUT)
 
                     if (!runner.signaled) {
+                        AgentNDK.log.error("ANR monitor signaled false, ANR detected")
                         notify()
                         runner.wait()
```
Comment (cthomas-newrelic) [inline agent-ndk/src/main/java/com/newrelic/agent/android/ndk/ANRMonitor.kt:53]: Why wait() here? That should cause the thread to block.

File: agent-ndk/src/main/java/com/newrelic/agent/android/ndk/ANRMonitor.kt [hunk 3]
```diff
@@ -152,6 +156,7 @@ open class ANRMonitor {
                                 exception.printStackTrace()
                             }
                         }
+
```
Comment (cthomas-newrelic) [inline agent-ndk/src/main/java/com/newrelic/agent/android/ndk/ANRMonitor.kt:159]: Has the runner been signaled here? If so, what happens on the next iteration?

File: agent-ndk/src/main/java/com/newrelic/agent/android/ndk/ANRMonitor.kt [hunk 4]
```diff
@@ -48,8 +48,12 @@ open class ANRMonitor {
                     runner.wait(ANR_TIMEOUT)
 
                     if (!runner.signaled) {
+                        AgentNDK.log.error("ANR monitor signaled false, ANR detected")
                         notify()
                         runner.wait()
+                    } else {
+                        AgentNDK.log.audit("ANR monitor signaled true, reset to false")
+                        runner.signaled = false
```
Comment (cthomas-newrelic) [inline agent-ndk/src/main/java/com/newrelic/agent/android/ndk/ANRMonitor.kt:56]: The boolean is updated, what about the object's signal/wait state?

File: agent-ndk/src/main/java/com/newrelic/agent/android/ndk/ANRMonitor.kt (hunk 2 above)
Comment (ywang-nr) [reply agent-ndk/src/main/java/com/newrelic/agent/android/ndk/ANRMonitor.kt:53]: This potion is part of the original code, we can remove it but what was the reason we had it before?

**Review by cthomas-newrelic — APPROVED** [review]
Consider an ANR counter like Splunk uses, but otherwise 🚢

**Pull request merged**